### PR TITLE
Replaces usage of device UUID with device name

### LINKED ISSUE #757: Revert CI change once Circle is fixed
Revert https://github.com/Moya/Moya/pull/753 once Circle has fixed its duplicate simulators issue.

### PR DESCRIPTION
**Pull request by pedrovereza**
Fixes #757

The issue on Circle CI [was fixed](https://discuss.circleci.com/t/os-x-build-image-update-xcode-8-2/9355) :tada:

I had to make a few changes to include not only the device name but also the OS version. Not sure if I took the best approach, suggestions are welcome :wink:

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -46,35 +46,40 @@ def sdks
...
+    tvos: "Apple TV 1080p"
   }
 end
 
-def open_simulator_and_sleep(uuid)
-  return if uuid.nil? # Don't need a sleep on macOS because it runs first.
-  sh "xcrun instruments -w '#{uuid}' || sleep 15"
+def device_os
+  return {
+    ios: "10.2",
```
Comment (scottrhoyt) [inline Rakefile]: Unless we want to ensure compatibility with specific version, is there an option to do OS `latest`?

Comment (pedrovereza) [inline Rakefile]: Oh, I didn't know about `latest`. Seems to be a good choice for the `xcodebuild` call.
Is there something similar for `instruments -w`? 

Actually: Do we really need the call to Instruments? 🤔

Comment (scottrhoyt) [inline Rakefile]: You can check #915 for an idea on how to launch the latest iOS simulator. Not sure if we still need the instruments call.

**Comment by scottrhoyt** [comment]
Another idea for how to make this run a little smoother is to swap out the `build test` [here](https://github.com/Moya/Moya/pull/922/files#diff-52c976fc38ed2b4e3b1192f8a8e24cffR117) for a `build-for-testing test`. I have noticed better results that way as it delays the call to open the simulator until the test target has already been built. Therefore you are less likely to have a timeout issue.

File: Rakefile (hunk 1 above)
Comment (pedrovereza) [inline Rakefile]: I tried using `latest` in another branch just for testing, didn't work on Circle CI:
https://circleci.com/gh/Moya/Moya/1191

Did I miss something?

**Comment by petester42** [comment]
For iPhones using the name might not always work. If the simulator has a watch tied to it the OS and the name will not be unique. You can have more than one. I'm guessing this works for iPhone 6s because the watches are now tied to an iPhone 7 simulator in circle. 

Let's just keep that in mind if something breaks in the future.

**Pull request merged**